obspy/core/stream.py

```diff
@@ -3101,6 +3134,205 @@ def _dummy_stream_from_string(s):
+                for cha_without_comp, components in sorted(
+                        common_channels.items()):
+                    if components == set(component_pair):
+                        channels_ = [cha_without_comp + comp
+                                     for comp in component_pair]
+                        self._rotate_specific_channels_to_zne(
+                            net, sta, loc, channels_, inventory)
+        return self
+
+    def _rotate_specific_channels_to_zne(
+            self, network, station, location, channels, inventory):
+        """
+        Rotate three explicitly specified channels to ZNE.
+
+        >>> from obspy import read, read_inventory
+        >>> st = read("/path/to/ffbx_unrotated_gaps.mseed")
+        >>> inv = read_inventory("/path/to/ffbx.stationxml")
+        >>> st._rotate_specific_channels_to_zne(
+        ...     "BW", "FFB1", "", ["HHZ", "HH1", "HH2"],
+        ...     inv)  # doctest: +ELLIPSIS
+        <obspy.core.stream.Stream object at 0x...>
+
+        :type network: str
+        :param network: Network code of channels that should be rotated.
+        :type station: str
+        :param station: Station code of channels that should be rotated.
+        :type location: str
+        :param location: Location code of channels that should be rotated.
+        :type channels: list
+        :param channels: The three channel codes of channels that should be
+            rotated.
+        :type inventory: :class:`~obspy.core.inventory.inventory.Inventory` or
+            :class:`~obspy.io.xseed.parser.Parser`
+        :param inventory: Inventory or Parser with metadata of channels.
+        """
+        from obspy.signal.rotate import rotate2zne
+        from obspy.core.inventory import Inventory, Network
+        from obspy.io.xseed import Parser
+
+        if isinstance(inventory, (Inventory, Network)):
+            metadata_getter = inventory.get_channel_metadata
+        elif isinstance(inventory, Parser):
+            # xseed Parser has everything in get_coordinates method due to
+            # historic reasons..
+            metadata_getter = inventory.get_coordinates
+        else:
+            msg = 'Wrong type for "inventory": {}'.format(str(type(inventory)))
+            raise TypeError(msg)
+        # build temporary stream that has only those traces that are supposed
+        # to be used in rotation
+        st = self.select(network=network, station=station, location=location)
+        st = (st.select(channel=channels[0]) + st.select(channel=channels[1]) +
+              st.select(channel=channels[2]))
+        # remove the original unrotated traces from the stream
+        for tr in st.traces:
+            self.remove(tr)
+        # cut data so that we end up with a set of matching pieces for the tree
+        # components (i.e. cut away any parts where one of the three components
+        # has no data)
+        st._trim_common_channels()
```

Comment:
+1 Would be nice to use this method for the other rotations, too

Reply:
Hmm.. to not hold up 1.1.0 any longer, I would propose to handle this in a follow-up issue. Would you mind opening one for 1.2.0?